### PR TITLE
Add mod query_and_stop_aws_ec2_instances_by_tag

### PR DESCRIPTION
**Pull request by bigdatasourav**
### Checklist
- [ ] Issue(s) linked

### CODE REVIEW

File: public_cloud/query_and_stop_aws_ec2_instance/query_and_stop_aws_ec2_instance.fp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        aws_ec2_instance
+      where
+        instance_state = 'running'
+        and tags ->> 'status' = 'decom'
+    EOQ
+  }
+
+  # Stop the EC2 instances
+  step "pipeline" "aws_ec2_instance_stop" {
+    depends_on = [step.query.list_ec2_instances]
```
Comment (e-gineer) [inline public_cloud/query_and_stop_aws_ec2_instance/query_and_stop_aws_ec2_instance.fp:33]: This `depends_on` is not required and should be removed.

File: public_cloud/query_and_stop_aws_ec2_instance/query_and_stop_aws_ec2_instance.fp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    for_each   = { for row in step.query.list_ec2_instances.rows : row.instance_id => row }
```
Comment (e-gineer) [inline public_cloud/query_and_stop_aws_ec2_instance/query_and_stop_aws_ec2_instance.fp:34]: Why can't we just do
```
for_each = step.query.list_ec2_instances.rows
```
?

and then use `each.value.instance_id` further below?

File: public_cloud/query_and_stop_aws_ec2_instance/query_and_stop_aws_ec2_instance.fp (hunk 1 above)
Comment (bigdatasourav) [reply public_cloud/query_and_stop_aws_ec2_instance/query_and_stop_aws_ec2_instance.fp]: Updated.

File: public_cloud/query_and_stop_aws_ec2_instance/query_and_stop_aws_ec2_instance.fp (hunk 2 above)
Comment (bigdatasourav) [reply public_cloud/query_and_stop_aws_ec2_instance/query_and_stop_aws_ec2_instance.fp]: Yes, we can use for-each in this way as well. Updated.

File: public_cloud/query_and_stop_aws_ec2_instance/README.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
+# Query and Stop AWS EC2 Instance
```
Comment (cbruno10) [inline public_cloud/query_and_stop_aws_ec2_instance/README.md:1]:
```suggestion
# Query and Stop AWS EC2 Instances by Tag
```

File: public_cloud/query_and_stop_aws_ec2_instance/README.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+Query and Stop AWS EC2 Instance based on the tag value.
```
Comment (cbruno10) [inline public_cloud/query_and_stop_aws_ec2_instance/README.md:3]:
```suggestion
Query a list of running AWS EC2 instances with the tag key-value pair `status: decom` using Steampipe and then stop them.
```

File: public_cloud/query_and_stop_aws_ec2_instance/mod.fp [hunk 5]
```diff
@@ -0,0 +1,12 @@
+mod "query_and_stop_aws_ec2_instance" {
```
Comment (cbruno10) [inline public_cloud/query_and_stop_aws_ec2_instance/mod.fp:1]:
```suggestion
mod "query_and_stop_aws_ec2_instances_by_tag" {
```

File: public_cloud/query_and_stop_aws_ec2_instance/mod.fp [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+  title         = "Query and Stop AWS EC2 Instance"
```
Comment (cbruno10) [inline public_cloud/query_and_stop_aws_ec2_instance/mod.fp:2]:
```suggestion
  title         = "Query and Stop AWS EC2 Instances by Tag"
```

File: public_cloud/query_and_stop_aws_ec2_instance/mod.fp [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+  description   = "Query and Stop AWS EC2 Instance based on the tag value."
```
Comment (cbruno10) [inline public_cloud/query_and_stop_aws_ec2_instance/mod.fp:3]:
```suggestion
  description   = "Query and stop AWS EC2 instance based on the `status` tag value."
```

File: public_cloud/query_and_stop_aws_ec2_instance/query_and_stop_aws_ec2_instance.fp [hunk 8]
```diff
@@ -0,0 +1,41 @@
+pipeline "query_and_stop_aws_ec2_instance" {
```
Comment (cbruno10) [inline public_cloud/query_and_stop_aws_ec2_instance/query_and_stop_aws_ec2_instance.fp:1]:
```suggestion
pipeline "query_and_stop_aws_ec2_instances_by_tag" {
```

File: public_cloud/query_and_stop_aws_ec2_instance/query_and_stop_aws_ec2_instance.fp [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+  title       = "Query and Stop AWS EC2 Instance"
```
Comment (cbruno10) [inline public_cloud/query_and_stop_aws_ec2_instance/query_and_stop_aws_ec2_instance.fp:2]:
```suggestion
  title       = "Query and Stop AWS EC2 Instances by Tag"
```

File: public_cloud/query_and_stop_aws_ec2_instance/query_and_stop_aws_ec2_instance.fp [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+  description = "Query and Stop AWS EC2 Instance based on the tag value."
```
Comment (cbruno10) [inline public_cloud/query_and_stop_aws_ec2_instance/query_and_stop_aws_ec2_instance.fp:3]:
```suggestion
  description = "Query and stop AWS EC2 instance based on the `status` tag value."
```

File: public_cloud/query_and_stop_aws_ec2_instance/query_and_stop_aws_ec2_instance.fp [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+    default     = var.aws_region
+  }
+
+  param "aws_cred" {
+    type        = string
+    description = "Name for AWS credential to use. If not provided, the default credential will be used."
+    default     = var.aws_cred
+  }
+
+  # List EC2 instances which need to stop
```
Comment (cbruno10) [inline public_cloud/query_and_stop_aws_ec2_instance/query_and_stop_aws_ec2_instance.fp:17]:
```suggestion
```

I think the code is self-documenting, so no need for the comment

File: public_cloud/query_and_stop_aws_ec2_instance/query_and_stop_aws_ec2_instance.fp [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+        instance_id
+      from
+        aws_ec2_instance
+      where
+        instance_state = 'running'
+        and tags ->> 'status' = 'decom'
+    EOQ
+  }
+
+  # Stop the EC2 instances
```
Comment (cbruno10) [inline public_cloud/query_and_stop_aws_ec2_instance/query_and_stop_aws_ec2_instance.fp:31]:
```suggestion
```
I think the code is self-documenting, so no need for the comment

File: public_cloud/query_and_stop_aws_ec2_instance/query_and_stop_aws_ec2_instance.fp [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+  step "pipeline" "aws_ec2_instance_stop" {
```
Comment (cbruno10) [inline public_cloud/query_and_stop_aws_ec2_instance/query_and_stop_aws_ec2_instance.fp:32]:
```suggestion
  step "pipeline" "stop_aws_ec2_instances" {
```

File: public_cloud/query_and_stop_aws_ec2_instance/query_and_stop_aws_ec2_instance.fp [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+    for_each = step.query.list_ec2_instances.rows
+    pipeline = aws.pipeline.stop_ec2_instances
+    args = {
+      region       = param.aws_region
+      cred         = param.aws_cred
+      instance_ids = ["${each.value.instance_id}"]
+    }
+  }
+}
```
Comment (cbruno10) [inline public_cloud/query_and_stop_aws_ec2_instance/query_and_stop_aws_ec2_instance.fp:41]: Can you please add an output that has an value like, `Stopped instances: i-123, i-456, i-789`

**Review by cbruno10 — CHANGES_REQUESTED** [review]
@bigdatasourav Please update filenames to match the new mod/pipeline name as well, thanks!

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**